### DISCUSSION TITLE
Including Leaders in Custom Reference

### DISCUSSION THREAD

**Discussion by pursellta** [milaboratory/mixcr#1596] [General]
Hello! 
I am trying to generate a custom reference for a non-model organism from my genome annotations. I wanted to include the leader sequences (i.e. Vtranscript) to improve the V-gene calling as my libraries are 5'RACE, but when I run the following code:
`mixcr buildLibrary --debug --v-genes-from-fasta IGHVwLeaders.fasta --v-gene-feature VTranscript --j-genes-from-fasta IGHJ.fasta --d-genes-from-fasta IGHD.fasta --c-genes-from-fasta IGHC.fasta --chain IGH --taxon-id 29078 --species epfu IGH-V2.json.gz`
It fails to extract the VTranscript. I have successfully generated and used a library using the VRegion as suggested in the documentation. I am using v4.6.0.
Thanks for your help!

**Comment by pursellta** [comment]
UPDATE: It still doesn't work when I change the feature to `VTranscriptWithout5UTR`  which more accurately describes my input sequence. That said, I was able to get it to work by altering the input to be from `L1Begin` to `VEnd` and using the feature `VGene` but it sometimes falsely puts some of the `L1` as `V5UTR`.

**Comment by mizraelson** [comment]
Hi,
Can you please forward this request to support@milaboratories.com
